payloadinterceptor.go

```diff
@@ -23,34 +26,55 @@ import (
 type streamingClientInterceptor struct {
 	connect.StreamingClientConn
 
-	receive func(any, connect.StreamingClientConn) error
-	send    func(any, connect.StreamingClientConn) error
-	onClose func()
+	ctx   context.Context //nolint:containedctx
+	state *state
 
+	requestStarted sync.Once
 	mu             sync.Mutex
 	requestClosed  bool
 	responseClosed bool
-	onCloseCalled  bool
+}
+
+func (s *streamingClientInterceptor) init() {
+	s.requestStarted.Do(func() {
+		header := s.RequestHeader()
+		s.ctx = s.state.start(s.ctx, header)
+	})
```

Comment:
This seems problematic. We've already passed a context to the `next` handler in the interceptor chain. So adding a span to the context here is incorrect, right? How will trace/span info get propagated on-the-wire and to the server if we already passed along the wrong context to those operations?